core/merge.js

```diff
@@ -349,6 +360,9 @@ class Merge {
 
       let existingDstRev = existingDstRevs[dst._id]
       if (existingDstRev && folder.overwrite) dst._rev = existingDstRev
+      if (side === 'remote' && newRevs && newRevs[dst.remote._id]) {
```

Comment:
Is the `side === 'remote'` check needed?
Wouldn't `newRevs` be null anyway? 
Maybe rename it to `remoteRevs` or `newRemoteRevs` everywhere to make it clear it's remote-specific?